app/onboarding/page.tsx

```
@@ -0,0 +1,179 @@
+  selectedGardenType,
+  setSelectedGardenType,
+}) => {
+  return (
+    <div>
+      <h2>What type of garden do you want to create?</h2>
+      {gardenTypes.map(type => (
+        <label key={type}>
+          <input
+            type="radio"
+            name="gardenType"
+            value={type}
+            checked={selectedGardenType === type}
+            onChange={e => setSelectedGardenType(e.target.value)}
+          />
+          {type}
+        </label>
+      ))}
+    </div>
+  );
+};
+
+// Step 3: Do you have a plot?
+const PlotSelection: React.FC<PlotSelectionProps> = ({
+  selectedPlot,
+  setSelectedPlot,
+}) => {
+  return (
+    <div>
+      <h2>Do you already have a plot?</h2>
+      {plotOptions.map(option => (
+        <label key={option.label}>
+          <input
+            type="radio"
+            name="plot"
+            value={String(option.value)}
+            checked={selectedPlot === option.value}
+            onChange={() => setSelectedPlot(option.value)}
+          />
+          {option.label}
+        </label>
+      ))}
+    </div>
+  );
+};
+
+// Main Onboarding Component
+const OnboardingFlow = () => {
+  const [step, setStep] = useState(1);
+  const [selectedState, setSelectedState] = useState<string>('');
+  const [selectedGardenType, setSelectedGardenType] = useState<string>('');
+  const [selectedPlot, setSelectedPlot] = useState<boolean>(false);
+
+  const handleNext = () => {
+    setStep(step + 1);
+  };
+
+  const handleBack = () => {
+    setStep(step - 1);
+  };
+
+  const handleSubmit = async () => {
+    const profile: Profile = {
+      user_id: id,
+      state: selectedState,
+      email: '',
+      phone_num: '',
+      user_type: selectedGardenType,
+      has_plot: selectedPlot,
+    };
+    try {
+      const updatedProfile = await upsertProfile(profile);
+      console.log('Profile successfully upserted:', updatedProfile);
+    } catch (err) {
+      console.error('Error upserting profile:', err);
```

let's throw an error instead!